### PR TITLE
Feature/inbox functionality

### PR DESCRIPTION
**Pull request by LeonelHays**
## Description
--(co-author Ryan Mark)(Driver-narrator)

Trello Link:  https://trello.com/c/ECXD68Le/170-create-the-endpoints-on-the-backend-to-support-inbox-messaging

Created the model and router for the full C.R.U.D. functionality for the inbox end-points.
Added middleware in end-points for errors and required inputs.
Connected the inbox Router to the App.js file.

## Loom Video

https://www.loom.com/share/ab48ab2d526a40699cacd096368a8856

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have removed unnecessary comments/console logs from my code
- [ ] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by CamMeiners — APPROVED** [review]

File: api/middleware/roleAuthentication.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
 const roleAuthentication = (...args) => (req, res, next) => {
   // role type is inside req.profile.body
   const { role } = req.profile;
+  console.log(role);
```
Comment (Ryan-E-Mark) [inline api/middleware/roleAuthentication.js]: Don't forget to remove the console.log

**Review by Ryan-E-Mark — APPROVED** [review]
Great work on getting this messaging router built out!

File: api/middleware/roleAuthentication.js (hunk 1 above)
Comment (mcmillanj) [reply api/middleware/roleAuthentication.js]: I see what Ryan said that the console log needs to be removed. I also see that changes were made to roleAuthentication.js that were fixes for the const {role} to const roles .good work

**Review by mcmillanj — APPROVED** [review]

**Review by mcmillanj — APPROVED** [review]

File: package-lock.json [hunk 2]
```diff
@@ -1,10776 +1,8 @@
 {
```
Comment (katherineyevsukov) [inline package-lock.json:1]: What is going on with the package-lock file being removed?

File: api/middleware/roleAuthentication.js [hunk 3]
```diff
@@ -1,8 +1,9 @@
...
   //check to see if role matcheds the role that have access to the endpoint
```
Comment (katherineyevsukov) [inline api/middleware/roleAuthentication.js]: I think comment can be removed, code is clear

File: api/middleware/roleAuthentication.js [hunk 4]
```diff
@@ -1,8 +1,9 @@
 const roleAuthentication = (...args) => (req, res, next) => {
   // role type is inside req.profile.body
```
Comment (katherineyevsukov) [inline api/middleware/roleAuthentication.js]: comment can be removed as code is clear, also the console.log

**Review by katherineyevsukov — CHANGES_REQUESTED** [review]
Things with the inbox router/model look good and nice job adding tests. Few things that need to be cleaned up and I am not sure what is going on with changes to package-lock.json file

**Pull request merged**